pytest_fixtures/broker.py

```diff
@@ -1,12 +1,43 @@
 import pytest
 from broker.broker import VMBroker
+from wait_for import wait_for
 
 from robottelo.constants import BROKER_RHEL77
 from robottelo.hosts import Capsule
 from robottelo.hosts import ContentHost
 from robottelo.hosts import Satellite
 
 
+@pytest.fixture(scope="session")
+def satellite_factory():
+    def factory(retry_limit=3, delay=300, **broker_args):
```

Comment:
I think the function signature and behavior can be simplified to just take a `timeout` and pass it to wait_for.

Reply:
I can go either way with this one. The question for me is more of whether I want users to have to keep in mind how long it takes to check out a Satellite.

Reply:
fixed

Reply:
I don't think they should have to consider that or that should be baked into this function's argument structure somehow. It should just be configurable for the amount of time it will wait.

The way you have it now makes an assumption about the amount of time it takes to get a satellite, and would have to be adjusted based on the performance of the broker provider, or things like resource pools cutting down deployment time.

The current default delay is very long, there is no need to wait 5 minutes between attempts to call checkout.

Callers of this function should be able to just pass along the arguments that wait_for uses.

Reply:
I'm going to disagree. It is an easier use model in the current form as the user doesn't need to determine their own estimate for average checkout times.
Adding a config for average checkout times is overkill for this.
A 5 minute delay may be overkill, but since there should only be one reason (SLA limit) for this to fail, 5 minutes may give enough time for another session to complete and perform a checkin.